### PR TITLE
Feature/ipehr 134 api get stored queries by ver

### PR DESCRIPTION
**Pull request by Setest**

### CODE REVIEW

**Review by SealTV — APPROVED** [review]
lgfm

File: src/pkg/api/stored_query.go [hunk 1]
```diff
@@ -51,6 +52,63 @@ func (h *QueryHandler) ListStored(c *gin.Context) {
 	c.JSON(http.StatusOK, queryList)
 }
 
+// Get
+// @Summary      Get list stored queries by version
```
Comment (1db1) [inline src/pkg/api/stored_query.go]: This method involves obtaining a single object, not a list

File: src/pkg/api/stored_query.go [hunk 2]
```diff
@@ -51,6 +52,63 @@ func (h *QueryHandler) ListStored(c *gin.Context) {
...
+// @Param        version                 path      string  false  "A SEMVER version number. This can be a an exact version (e.g. 1.7.1), or a pattern as partial prefix, in a form of {major} or {major}.{minor} (e.g. 1 or 1.0), in which case the highest (latest) version matching the prefix will be considered."
+// @Param        Authorization           header    string  true  "Bearer AccessToken"
+// @Param        AuthUserId              header    string  true  "UserId UUID"
+// @Success      200                     {object}  []model.StoredQuery
+// @Failure      400                               "Is returned when the request has invalid content."
+// @Failure      404                               "Is returned when a stored query with {qualified_query_name} and {version} does not exist."
+// @Failure      500                               "Is returned when an unexpected error occurs while processing a request"
+// @Router       /definition/query/{qualifiedQueryName}/{version} [get]
+func (h *QueryHandler) ListStoredVersion(c *gin.Context) {
```
Comment (1db1) [inline src/pkg/api/stored_query.go]: ListStoredVersion -> GetStoredByVersion

File: src/pkg/api/stored_query.go [hunk 3]
```diff
@@ -51,6 +52,63 @@ func (h *QueryHandler) ListStored(c *gin.Context) {
...
+// @Description  Retrieves the definition of a particular stored query (at specified version) and its associated metadata.
+// @Description  https://specifications.openehr.org/releases/ITS-REST/latest/definition.html#tag/Query/operation/definition_query_list
+// @Tags         QUERY
+// @Accept       json
+// @Produce      json
+// @Param        qualified_query_name    path      string  false  "If pattern should given be in the format of [{namespace}::]{query-name}, and when is empty, it will be treated as "wildcard" in the search."
+// @Param        version                 path      string  false  "A SEMVER version number. This can be a an exact version (e.g. 1.7.1), or a pattern as partial prefix, in a form of {major} or {major}.{minor} (e.g. 1 or 1.0), in which case the highest (latest) version matching the prefix will be considered."
+// @Param        Authorization           header    string  true  "Bearer AccessToken"
+// @Param        AuthUserId              header    string  true  "UserId UUID"
+// @Success      200                     {object}  []model.StoredQuery
```
Comment (1db1) [inline src/pkg/api/stored_query.go]: []model.StoredQuery -> model.StoredQuery

File: src/pkg/api/stored_query.go [hunk 4]
```diff
@@ -51,6 +52,63 @@ func (h *QueryHandler) ListStored(c *gin.Context) {
...
+	// This does not satisfy the specification. https://specifications.openehr.org/releases/ITS-REST/latest/definition.html#tag/Query/operation/definition_query_list
+	// But otherwise it is not clear how the client can get the full list of stored queries
+	qName := c.Param("qualifiedQueryName")
+	if qName == "" {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "qualified_query_name is empty"})
+		return
+	}
+
+	sq, err := h.service.GetByVersion(c, userID, qName, v)
+
```
Comment (1db1) [inline src/pkg/api/stored_query.go]: extra blank line

File: src/pkg/api/stored_query_test.go [hunk 5]
```diff
@@ -111,6 +112,123 @@ func TestStoredQueryHandler_List(t *testing.T) {
 	}
 }
 
+func TestStoredQueryHandler_ListVersion(t *testing.T) {
```
Comment (1db1) [inline src/pkg/api/stored_query_test.go]: ListVersion -> GetByVersion

**Review by 1db1 — CHANGES_REQUESTED** [review]

File: src/pkg/api/stored_query.go [hunk 6]
```diff
@@ -51,6 +52,63 @@ func (h *QueryHandler) ListStored(c *gin.Context) {
...
+	v, err := base.NewVersionTreeID(version)
+	if err != nil {
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	// Here we do not check the existence of the argument.
+	// This does not satisfy the specification. https://specifications.openehr.org/releases/ITS-REST/latest/definition.html#tag/Query/operation/definition_query_list
+	// But otherwise it is not clear how the client can get the full list of stored queries
```
Comment (1db1) [inline src/pkg/api/stored_query.go]: This comment is not relevant here

**Review by 1db1 — APPROVED** [review]

**Pull request merged**